File: backend/app/src/main/kotlin/net/animeta/backend/controller/v2/ChartController.kt
package net.animeta.backend.controller.v2

import net.animeta.backend.service.ChartService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v2/charts")
class ChartController(
    private val chartService: ChartService
) {
    @GetMapping("/works/weekly")
    fun getWeeklyWorks(@RequestParam limit: Int) = chartService.getWeeklyWorks(limit)
}

File: backend-ts/src/controllers/chart.controller.ts
import { Controller, Get, ParseIntPipe, Query } from "@nestjs/common";
import { ChartItem, ChartItemWork } from 'shared/types';
import { ChartService } from "src/services/chart.service";

@Controller('/api/v4/charts')
export class ChartController {
  constructor(
    private chartService: ChartService,
  ) {}

  @Get('works/weekly')
  async getWeeklyWorks(
    @Query('limit', new ParseIntPipe()) limit: number
  ): Promise<Array<ChartItem<ChartItemWork>>> {
    return this.chartService.getWeeklyWorks(limit)
  }
}
